gnome-image-installer/pages/diskimage/gis-diskimage-page.c

```diff
@@ -103,34 +104,26 @@ gis_diskimage_page_selection_changed(GtkWidget *combo, GisPage *page)
   gis_store_set_object (GIS_STORE_IMAGE, G_OBJECT (file));
   if (g_str_has_suffix (image, ".gz"))
     {
-      gint64 size = get_gzip_disk_image_size (image);
-      if (size <= 0)
-        {
-          size = 1*1024*1024*1024;
-          size *= 8;
-        }
-      gis_store_set_required_size (size);
+      required_size = get_gzip_disk_image_size (image);
     }
   else if (g_str_has_suffix (image, ".xz"))
     {
-      gint64 size = get_xz_disk_image_size (image);
-      if (size <= 0)
-        {
-          size = 1*1024*1024*1024;
-          size *= 8;
-        }
-      gis_store_set_required_size (size);
+      required_size = get_xz_disk_image_size (image);
     }
   else if (g_str_has_suffix (image, ".img") || g_strcmp0 (image, live_device_path) == 0)
     {
-      gint64 size = get_disk_image_size (image);
-      if (size <= 0)
-        {
-          size = 1*1024*1024*1024;
-          size *= 8;
-        }
-      gis_store_set_required_size (size);
+      required_size = get_disk_image_size (image);
+    }
+
+  /* This should not happen. Before adding the image to the model, we checked
+   * the filename and validated the GPT header.
+   */
+  if (required_size == 0)
+    {
+      g_warn_if_fail (required_size > 0);
+      required_size = G_MAXINT64;
```

Comment:
This will now warn if the `image` name doesn’t match any of the suffix checks above. Are you confident that won’t be the case? (If so, maybe add an `else { g_assert_not_reached (); }`?)